UFile.go

```diff
@@ -0,0 +1,276 @@
+package ucloud
+
+import (
+	"bytes"
+	"crypto/hmac"
+	"crypto/sha1"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+func (u *UcloudApiClient) SignatureUFile(privateKey string, stringToSign string) string {
+	mac := hmac.New(sha1.New, []byte(privateKey))
+	mac.Write([]byte(stringToSign))
+	return base64.StdEncoding.EncodeToString(mac.Sum(nil))
+}
+
+type SignParam struct {
+	HttpVerb                   string
+	ContentMd5                 string
+	ContentType                string
+	Date                       string
+	CanonicalizedUCloudHeaders string
+	CanonicalizedResource      string
+}
+
+func (param *SignParam) String() string {
+	return param.HttpVerb + "\n" +
+		param.ContentMd5 + "\n" +
+		param.ContentType + "\n" +
```

Comment:
thanks for reviewing my code, since this is not the newest verion. please
check: http:github.com/himyxia/ucloud-go-sdk if you are interested